lib/rubocop/cop/style/multiline_list.rb

```diff
@@ -0,0 +1,158 @@
+      #     # good with EnableMethodParams
+      #     def method(
+      #         foo, bar,
+      #         baz)
+      #       do_something
+      #     end
+      #
+      #     # ignored with EnableMethodParams
+      #     def method foo,
+      #         bar
+      #       do_something
+      #     end
+      #
+      #     # bad with EnableMethodArgs
+      #     method(foo, bar,
+      #       baz)
+      #
+      #     # good with EnableMeethodArgs
+      #     method(
+      #       foo, bar,
+      #       baz)
+      #
+      #     # ignored with EnableMethodArgs
+      #     method foo, bar,
+      #       baz
+      #
+      #     # bad with EnableHashElements
+      #     { a: 1,
+      #       b: 2}
+      #
+      #     # good with EnableHashElements
+      #     {
+      #       a: 1,
+      #       b: 2 }
+      #
+      #     # bad with EnableArrayElements
+      #     [ :a,
+      #       :b]
+      #
+      #     # good with EnableArrayElements
+      #     [
+      #       :a,
+      #       :b]
+      #
+      class MultilineList < Cop
+        include OnMethodDef
+
+        MSG = 'Add a line break before the first element of a multi-line list.'
+
+        def on_method_def(node, _method_name, args, _body)
+          return unless cop_config['EnableMethodParams']
+
+          check_method(node, args.to_a)
+        end
+
+        def on_send(node)
+          return unless cop_config['EnableMethodArgs']
+
+          _receiver, _name, *args = *node
+
+          check_method(node, args)
+        end
+
+        def on_array(node)
+          return unless cop_config['EnableArrayElements']
+
+          return if !node.loc.begin && !assignment_on_same_line?(node)
+
+          check_children(node, node.children)
+        end
+
+        def on_hash(node)
+          return unless cop_config['EnableHashElements']
+
+          if node.loc.begin
+            check_children(node, node.children)
+          elsif !method_uses_parens?(node.parent, node)
+            return
+          else
+            check_children(node, node.children, node.parent)
+          end
+        end
+
+        def autocorrect(node)
+          ->(corrector) { corrector.insert_before(node.loc.expression, "\n") }
+        end
+
+        private
+
+        def check_method(node, children)
+          return if children.empty?
+
+          return unless method_uses_parens?(node, children.first)
+
+          check_children(node, children)
+        end
+
+        def assignment_on_same_line?(node)
+          source = node.loc.expression.source_line[0...node.loc.column]
+          source =~ /\s*\=\s*$/
+        end
+
+        def method_uses_parens?(node, limit)
+          source = node.loc.expression.source_line[0...limit.loc.column]
+          source =~ /\s*\(\s*$/
+        end
+
+        def check_children(node, children, start = node)
+          return if children.size < 2
+
+          line = start.loc.line
+          min = children.min_by { |n| n.loc.line }
```

Comment:
Assuming that the node has a `begin` and an `end`, which I think it will in this case, you can use `node.loc.begin.line` and `node.loc.end.line`. The same will hold true for other attributes off of `loc`.